test/e2e/throughputanomalydetection_test.go

```diff
@@ -288,6 +379,8 @@ func taddeleteJob(t *testing.T, data *TestData, jobName string) (stdout string,
 }
 
 func tadretrieveJobResult(t *testing.T, data *TestData, jobName string) (stdout string, err error) {
+	mutex.Lock()
+	defer mutex.Unlock()
```

Comment:
I'm a little bit lost here. Why we need the lock in each tad function? Isn't the e2e test series?

Reply:
That will be because there will always be 3 cases which are running in parallel.